http/src/main/java/io/micronaut/http/MediaType.java

```diff
@@ -817,27 +818,62 @@ public static MediaType forFilename(String filename) {
         return MediaType.TEXT_PLAIN_TYPE;
     }
 
+    /**
+     * Resolve the {@link MediaType} for the given media/content type text.
+     *
+     * @param mediaType The media/content type text
+     * @return An {@link Optional} of {@link MediaType}
+     */
+    public static Optional<MediaType> forMediaType(String mediaType) {
```

Comment:
Since the mime file has more definitions than the switch case, this can be utilized to access that file's definitions instead. I use it in the FileUpload implementations to make them keep using the browser provided file type.